driver-sync/src/main/com/mongodb/client/gridfs/GridFSUploadStreamImpl.java

```diff
@@ -136,6 +144,13 @@ public void write(final byte[] b, final int off, final int len) {
         }
     }
 
+    private void checkTimeout() {
+        if (timeout != null && timeout.hasExpired()) {
+            // TODO (CSOT) - JAVA-5248 Update to MongoOperationTimeoutException
+            throw new MongoExecutionTimeoutException(TIMEOUT_MESSAGE);
```

Comment:
Nit: Use `TimeoutContext.createMongoTimeoutException` then we have a single place to manage all timeout based exceptions.